src/gen_rules.ml

```diff
@@ -1225,6 +1225,32 @@ module Gen(P : Params) = struct
          ~dir
          ~targets)
 
+  let alias_rules (alias_conf : Alias_conf.t) ~dir =
+    let digest =
+      let source =
+        match alias_conf.action with
+        | None -> ""
+        | Some a -> Sexp.to_string (User_action.Unexpanded.sexp_of_t a) in
```

Comment:
We should have:

```ocaml
| None -> "none"
| Some a -> "some" ^ ...
```

otherwise no action and `""` would produce the same digest. `""` would fail anyway since it is not a valid bash command, but you could get an error about multiple rules generated for a weird target, which is not great. Thinking about it, we should probably include the dependencies in the digest, otherwise we would get a clash in this case:

```scheme
(alias
  ((name runtest)
   (deps (prog.exe file1)
   (action "./${*}")))

(alias
  ((name runtest)
   (deps (prog.exe file2)
   (action "./${*}")))
```